sound/soc/sof/intel/hda-loader.c

```diff
@@ -396,12 +396,16 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 		dev_dbg(sdev->dev, "IMR restore supported, booting from IMR directly\n");
 		hda->boot_iteration = 0;
 		ret = hda_dsp_boot_imr(sdev);
-		if (!ret)
+		if (!ret) {
+			hda->booted_from_imr = true;
```

Comment:
this is surprising, we already skip the base firmware loading when booting from IMR today, not sure why we need the same information stored again for libraries?

Reply:
I don't want to re-use the same flag for the decision to **do** or **not do** an IMR boot vs if we **booted** or **not booted** from IMR.
IMR boot can fail and then we will try a cold boot. I'm certain that we will bump into issues when re-purposing signals.

Reply:
that's fine, maybe the definition needs to be a bit clearer so that it's clear that it's just a record of what happened during boot, with the decision depending on other criteria. I clearly missed the nuance when reading the code.

Reply:
The commit message in https://github.com/thesofproject/linux/pull/3826/commits/b49f6c4ca971efa973d31fd9162c70b7b03df8c8 should be expanded?
Or shall we have comment? I think I have it when checking the flag, but not when setting it.

Reply:
yes, commit message is fine.